manager/allocator/networkallocator/networkallocator.go

```diff
@@ -186,12 +187,28 @@ func (na *NetworkAllocator) ServiceAllocate(s *api.Service) (err error) {
 		return
 	}
 
-	// First allocate VIPs for all the pre-populated endpoint attachments
-	for _, eAttach := range s.Endpoint.VirtualIPs {
-		if err = na.allocateVIP(eAttach); err != nil {
-			return
+	// Allocate VIPs for all the pre-populated endpoint attachments
+	eVIPs := s.Endpoint.VirtualIPs[:0]
```

Comment:
What is the advantage of defining `eVIPs` this way compared to ` var eVIPs []*api.Endpoint_VirtualIP` ?
Is it so you do not need to care what types the elements are and yo udo not need to import their package ?